classes/Http/Action/Signup/ProcessAction.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Http\Action\Signup;
+
+use Illuminate\Container\Container;
+use Illuminate\Filesystem\Filesystem;
+use Illuminate\Translation;
+use Illuminate\Validation;
+use OpenCFP\Domain\Services;
+use OpenCFP\Domain\ValidationException;
+use Symfony\Component\HttpFoundation;
+use Symfony\Component\Routing;
+use Twig_Environment;
+
+final class ProcessAction
+{
+    /**
+     * @var Services\Authentication
+     */
+    private $authentication;
+
+    /**
+     * @var Services\AccountManagement
+     */
+    private $accounts;
+
+    /**
+     * @var Routing\Generator\UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    public function __construct(
+        Services\Authentication $authentication,
+        Services\AccountManagement $accounts,
+        Twig_Environment $twig,
+        Routing\Generator\UrlGeneratorInterface $urlGenerator
+    ) {
+        $this->authentication = $authentication;
+        $this->accounts       = $accounts;
+        $this->urlGenerator   = $urlGenerator;
+    }
+
+    public function __invoke(HttpFoundation\Request $request): HttpFoundation\Response
+    {
+        try {
+            $this->validate($request, [
```

Comment:
Can't write any unit tests as the validation facilities are created within this class.